include/cantera/thermo/ThermoFactory.h

```diff
@@ -107,6 +107,24 @@ inline ThermoPhase* newThermoPhase(const std::string& model)
     return ThermoFactory::factory()->create(model);
 }
 
+//! Create a new ThermoPhase instance.
+/*!
+ * @param model   String to look up the model against
+ * @returns a shared pointer to a new ThermoPhase instance matching the model string.
+ *   Throws an exception UnknownThermoPhaseModel if the string wasn't matched; or a
+ *   CanteraError if something else went wrong.
+ */
+inline shared_ptr<ThermoPhase> newThermo(const std::string& model)
+{
+    ThermoPhase* tptr = ThermoFactory::factory()->create(model);
+    if (tptr == 0) {
+        CanteraError(
+            "newThermo", "Encountered exception while creating thermo "
+            "model '{}'.", model);
+    }
```

Comment:
I don't think this check is necessary, though I do see some documentation that incorrectly suggests otherwise. Any exception raised in the `create` method will just propagate through, and I don't think there's any case where it will return a null pointer.

Reply:
tried to retain what I found elsewhere. But this makes sense.